fog/overseer/server/src/worker.rs

```diff
@@ -137,23 +137,28 @@ where
     pub fn start(
         ingest_clients: Vec<FogIngestGrpcClient>,
         recovery_db: DB,
-        stop_requested: Arc<AtomicBool>,
+        is_enabled: Arc<AtomicBool>,
         logger: Logger,
     ) {
         let thread = Self {
             ingest_clients,
             recovery_db,
-            stop_requested,
+            is_enabled,
             logger,
         };
         thread.run();
     }
 
     fn run(self) {
-        while !self.stop_requested.load(Ordering::SeqCst) {
+        loop {
```

Comment:
The thread is never going to terminate if the loop doesn't have an exit condition.

Reply:
Fixed in most recent commit.